test/integ/axpy/src/axpy.cpp

```diff
@@ -30,25 +30,24 @@ class AxpyKernel
     //! \tparam TAcc The type of the accelerator the kernel is executed on..
     //! \tparam TElem The matrix element type.
     //! \param acc The accelerator the kernel is executed on.
-    //! \param numElements Specifies the number of elements of the vectors X and Y.
     //! \param alpha Scalar the X vector is multiplied with.
     //! \param X Vector of at least n elements.
     //! \param Y Vector of at least n elements.
     ALPAKA_NO_HOST_ACC_WARNING
-    template<typename TAcc, typename TElem, typename TIdx>
+    template<typename TAcc, typename TMemoryHandle, typename TElem, typename TIdx>
     ALPAKA_FN_ACC auto operator()(
         TAcc const& acc,
-        TIdx const& numElements,
         TElem const& alpha,
-        TElem const* const X,
-        TElem* const Y) const -> void
+        alpaka::Accessor<TMemoryHandle, TElem, TIdx, 1, alpaka::ReadAccess> const X,
+        alpaka::Accessor<TMemoryHandle, TElem, TIdx, 1, alpaka::ReadWriteAccess> const Y) const -> void
```

Comment:
We need different `TMemoryHandle`s for different access modes (which is why I want to get rid of `TMemoryHandle` in the interface).

Reply:
Do we? `TMemoryHandle` is `float*` here in both cases. Do you have different types in SYCL when the access mode is different?

Reply:
Yup, similar to our accessors: `sycl::accessor<int, 1, sycl::access::mode::read_write, /* more params */>`.

Reply:
How do you deal with `alpaka::Accessor<..., std::tuple<alpaka::ReadAccess, alpaka::ReadWriteAccess>>` then? This was my reason to use `T*` consistently for `ViewAccessor`s, independent of the access mode.

Reply:
As you know I'm not a big fan of multiple access modes ;-) I'm still trying to figure out the best way to do this. I'm leaning towards passing `read_write` SYCL accessors internally if there is a tuple of access modes and just change the alpaka access mode for the resulting alpaka accessor.